source/extensions/http/stateful_session/cookie/cookie.h

```diff
@@ -69,17 +69,19 @@ class CookieBasedSessionStateFactory : public Envoy::Http::SessionStateFactory {
     envoy::Cookie cookie;
     if (cookie.ParseFromString(decoded_value)) {
       address = cookie.address();
-      if (address.empty() || (cookie.expires() == 0)) {
+      if (address.empty()) {
```

Comment:
I wonder if this needs a runtime-flag.
I tend to no (IIUC this was the prior behavior and it is fixing an issue added in a prior PR, and it matches the description of the feature before and after that PR), but will keep it as a question for the next reviewer.

Reply:
For context, this is breaking one of our customers because the change happened during an upgrade. I'm planning to backport this. I'd argue we don't need a flag for a backported change.